### PR TITLE
[Fixes #294] Adds smarter paginator and server side good/bad/unvetted…

### LINKED ISSUE #294: Better pagination and filter applying
Need to make sure pagination is not overflowing when displayed (ellipsis in between?)
pag-overflow

Also related to this filtering (e.g. on observations page) is only applied to current page and not on all results (and then paginated again)

### PR DESCRIPTION
**Pull request by faulteh**
… filtering to Observation list view
The paginator will now do << 1 .. X Y Z .. L >> if there are lots of pages.

Previously the good/bad/unvetted selections only 'collapsed' the current page, and also would not recompute the pagination. Moved the filtering of good/bad/unvetted to inside the python view so changing the state of the buttons will send a request to django and return a new result. Current filter request is also kept within paginated results.

However, since has_no_data and has_verified_data etc are properties of the model rather than fields these can't be filtered with the Django database filter and instead done by list comprehensions. Might not scale, but I don't know if it could be done with Django F() and Q() functions.

### CODE REVIEW

**Comment by ppapadeas** [comment]
@faulteh I checked-out your branch and although I have 1000+ obs as local fixtures I get no results in /observations/ with no errors on console :S 
review_264

File: network/base/models.py [hunk 1]
```diff
@@ -259,7 +260,7 @@ class Data(models.Model):
     vetted_datetime = models.DateTimeField(null=True, blank=True)
     vetted_user = models.ForeignKey(User, related_name="vetted_user_set", null=True, blank=True)
     vetted_status = models.CharField(choices=OBSERVATION_STATUSES,
-                                     max_length=10, default='unknown')
+                                     max_length=20, default='unknown')
```
Comment (faulteh) [inline network/base/models.py]: The 'data_not_verified' choice would not fit in 10 chars and was getting truncated.

File: network/base/models.py [hunk 2]
```diff
@@ -232,17 +232,18 @@ def has_submitted_data(self):
     # observaton has at least 1 payload that has been verified good
     @property
     def has_verified_data(self):
-        return self.data_set.filter(vetted_status='verified').count()
+        return self.data_set.filter(vetted_status=u"('verified', 'Verified')").count()
```
Comment (faulteh) [inline network/base/models.py]: These filters were not working unless I used the choices tuple in a unicode quote!

File: network/templates/base/paginator.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{% load url_replace %}
+<nav>
+  <ul class="pagination">
+  {% if page_obj.has_previous %}
```
Comment (comzeradd) [inline network/templates/base/paginator.html]: You should indent here. Same way you do in the rest of the template.

**Pull request merged**